### ISSUE TITLE
(Incompatible) kernel option change

### ISSUE THREAD

**Issue by namhyung** [namhyung/uftrace#2]
After a discussion with @honggyukim I decided to change some kernel options:

Now `--kernel-full` (`-K`) option does 4 things:
 * set kernel depth to max when record
 * set number of writer threads to same as cpus (unless `--num-thread` option is given) when record
 * set internal timeout to 10us for checking kernel buffer when record
 * show kernel functions outside of fuser functions when replay (inverse of `--kernel-skip-out`)

This is not good.  I'd like to keep kernel tracing option more simple.  While the `-k` (or `--kernel`) option does the thing, I think I need to break the `-K` option too.

1. setting kernel depth to max is not practical for most cases IMHO and it produces many LOST records.  I recommend users to use a much lower kernel depth (default is 1 but you may want 2 or 3) unless you're a kernel developer.

2. default number of writer thread when tracing kernel should be determined depending on how much data it'll generate.  So I think it should consider the kernel depth limit rather than the value of this option.

3. The timeout of 10us seems so small and I'm afraid it may hurts the performance.

4. I guess that most users of uftrace focus on their code and how the kernel affect it during the execution.  So I think it'd be better hiding kernel functions outside of user functions by default.

Now I'm thinking that the `--kernel-full` option should do a single job.  The item 1 above can be done with `--kernel-depth` option and I think it'll be used more frequently than others.  So the short option `-K` now goes to the `--kernel-depth` instead of `--kernel-full` so it'll require an integer argument.  The item 2 will not require any changes of command line options.  The item 3 should be gone IMHO.  The last, item 4 is what `--kernel-full` will do from now on.

Once it merges the code in `review/kernel-option-change` branch (c45b247):

```
## simple kernel tracing
$ sudo uftrace -k hello

## increase kernel-depth to 2
$ sudo uftrace -k -K2 hello

## show all kernel functions (including outside of user functions)
$ sudo uftrace -k --kernel-full hello
```

**Comment by honggyukim** [comment]
Thanks for the summary. But I'd like to add one more comment on it.
Why don't we give '-k' option implicitly when either '-K' or '--kernel-full' option is enabled?

**Comment by namhyung** [comment]
Ok.  I'll push the updated version.

**Comment by honggyukim** [comment]
Thanks!

**Comment by namhyung** [comment]
Merged: 5c4cd8d
